lib/chat/views/chatroom/bloc/chat_action_bloc/chat_action_bloc.dart

```diff
@@ -0,0 +1,602 @@
+import 'package:firebase_database/firebase_database.dart';
+import 'package:equatable/equatable.dart';
+import 'package:bloc/bloc.dart';
+import 'package:likeminds_chat_fl/likeminds_chat_fl.dart';
+import 'package:likeminds_flutter_sample/chat/likeminds_chat_mm_fl.dart';
+import 'package:likeminds_flutter_sample/chat/service/likeminds_service.dart';
+import 'package:likeminds_flutter_sample/chat/service/service_locator.dart';
+import 'package:likeminds_flutter_sample/chat/utils/local_preference/local_prefs.dart';
+import 'package:likeminds_flutter_sample/chat/utils/realtime/realtime.dart';
+import 'package:likeminds_flutter_sample/chat/service/media_service.dart';
+import 'package:likeminds_flutter_sample/chat/views/chatroom/chatroom_components/Poll/constants/string_constant.dart';
+import 'package:likeminds_flutter_sample/chat/views/media/widget/media_helper_widget.dart';
+
+part 'chat_action_event.dart';
+part 'chat_action_state.dart';
+
+class ChatActionBloc extends Bloc<ChatActionEvent, ChatActionState> {
+  MediaService mediaService = MediaService(!isDebug);
+  final DatabaseReference realTime = LMRealtime.instance.chatroom();
+  int? lastConversationId;
+
+  ChatActionBloc() : super(ConversationInitial()) {
+    on<NewConversation>(
+      (event, emit) {
+        int chatroomId = event.chatroomId;
+        lastConversationId = event.conversationId;
+
+        realTime.onValue.listen(
+          (event) {
+            if (event.snapshot.value != null) {
+              final response = event.snapshot.value as Map;
+              final conversationId =
+                  int.parse(response["collabcard"]["answer_id"]);
+              if (lastConversationId != null &&
+                  conversationId != lastConversationId) {
+                add(UpdateConversationList(
+                  chatroomId: chatroomId,
+                  conversationId: conversationId,
+                ));
+              }
+            }
+          },
+        );
+      },
+    );
+    on<PostPollConversation>((event, emit) async {
+      try {
+        User user = UserLocalPreference.instance.fetchUserData();
+        DateTime currentDate = DateTime.now();
+        Conversation conversation = Conversation(
```

Comment:
Also, created these objects in some other class or file to main readability of this code.